server/internal/router/router.go

```diff
@@ -0,0 +1,192 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/server/internal/config"
+	"github.com/llm-operator/inference-manager/server/internal/k8s"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const tickPeriod = 60 * time.Second
+
+type model struct {
+	id string
+}
+
+// R manages the routing table of models to engines.
+type R struct {
+	// m is the route map.
+	m             *routeMap
+	engineClients map[string]v1.InferenceEngineInternalServiceClient
+	imeConfig     config.InferenceManagerEngineConfig
+}
+
+// New creates a new router.
+func New(c config.InferenceManagerEngineConfig) *R {
+	return &R{
+		m:             newRouteMap(),
+		engineClients: make(map[string]v1.InferenceEngineInternalServiceClient),
+		imeConfig:     c,
+	}
+}
+
+// Run updates the routes periodically and upon pod events.
+func (r *R) Run(ctx context.Context, errCh chan error) error {
+	coreClient, err := k8s.NewCoreClient()
+	if err != nil {
+		return fmt.Errorf("create K8sClient: %v", err)
+	}
+
+	stopper := make(chan struct{})
+	defer close(stopper)
+	defer runtime.HandleCrash()
+
+	engineLabels := map[string]string{
+		r.imeConfig.LabelKey: r.imeConfig.LabelValue,
+	}
+
+	// initialize the routes before starting the informer.
+	if err := r.refreshRoutes(ctx, coreClient, r.imeConfig.Namespace, engineLabels); err != nil {
```

Comment:
We don't need to pass `r.imeConfig.Namespace` to this function?

Reply:
`ListPods` need `namespace`

Reply:
But, `refreshRoutes` is a receiver of `r`, so it can just get from `r.imeConfig.Namespace`?